src/cljs/harja/views/urakka/toteumat/tiemerkinta_muut_tyot.cljs

```diff
@@ -0,0 +1,134 @@
+            [harja.ui.napit :as napit])
+  (:require-macros [cljs.core.async.macros :refer [go]]
+                   [reagent.ratom :refer [reaction run!]]
+                   [harja.atom :refer [reaction-writable]]))
+
+(defn- valinnat [e! {:keys [valittu-urakka valittu-sopimusnumero
+                            valitse-sopimusnumero valitun-urakan-hoitokaudet
+                            valittu-hoitokausi valitse-hoitokausi]}]
+  (let [muokkausoikeus? true] ;; TODO OIKEUSTARKISTUS
+    [:span
+     (valinnat/urakan-sopimus-ja-hoitokausi
+       valittu-urakka
+       valittu-sopimusnumero valitse-sopimusnumero
+       valitun-urakan-hoitokaudet valittu-hoitokausi valitse-hoitokausi)
+     [napit/uusi "Lisää toteuma" #(e! (tiedot/->UusiToteuma))
+      {:disabled (not muokkausoikeus?)}]]))
+
+(defn muu-tyo-lomake [e! tila {:keys [valittu-urakka valittu-sopimusnumero] :as riippuvuudet}]
+  (let [vanha-toteuma? (get-in tila [:valittu-toteuma :id])
+        valinnat (:valinnat tila)
+        muokkausoikeus? true] ;; TODO OIKEISTARKISTUS
+    (log "TILA:" (pr-str tila))
+    [:div
+     [napit/takaisin "Takaisin toteumaluetteloon"
+      #(e! (tiedot/->ValitseToteuma nil))]
+     [lomake {:otsikko      (if vanha-toteuma?
+                              "Muokkaa toteumaa"
+                              "Luo uusi toteuma")
+              :luokka       :horizontal
+              :voi-muokata? muokkausoikeus?
+              :muokkaa!     #(e! (tiedot/->MuokkaaToteumaa %))
+              :footer       [napit/palvelinkutsu-nappi
+                             "Tallenna toteuma"
+                             #(tiedot/tallenna-toteuma {:toteuma              (:valittu-toteuma tila)
+                                                        :urakka               (:id valittu-urakka)
+                                                        :sopimus              (first valittu-sopimusnumero)
+                                                        :alkupvm              (first (:sopimuskausi valinnat))
+                                                        :loppupvm             (second (:sopimuskausi valinnat))
+                                                        :uusi-laskentakohde   (:uusi-laskentakohde tila)}
+                                                       (:laskentakohteet tila))
+                             {:luokka       "nappi-ensisijainen"
+                              :ikoni        (ikonit/tallenna)
+                              :kun-onnistuu #(e! (tiedot/->ToteumaTallennettu %))
+                              :disabled     (or (not (lomake/voi-tallentaa? (:valittu-toteuma tila)))
+                                                (not muokkausoikeus?))}]}
+
+      [{:otsikko "Päivämäärä" :nimi :pvm :tyyppi :pvm :pakollinen? true}
+       {:otsikko "Hinta" :nimi :hinta :tyyppi :positiivinen-numero :pakollinen? true}
+       {:otsikko "Ylläpitoluokka" :nimi :yllapitoluokka :tyyppi :positiivinen-numero
+        :huomauta [[:yllapitoluokka]]}
```

Comment:
Ehkä sitä voidaan nyt tiukentaa. YHA API vaatii vain integerin, mutta vaaditaan me 1,2 tai 3 tai ei arvoa, ja hoidetaan sitä kautta oikea integer YHA API:in.